### PR TITLE
[# 111989775] As Ruote, so I don't keep being the worst offender in the entire slow_logs table, migrate my documents table to be indexable on "at"

### PR DESCRIPTION
**Pull request by soonahn**
https://github.com/ConsultingMD/frick/pull/223

[Tracker #111989775](https://www.pivotaltracker.com/story/show/111989775)

### CODE REVIEW

**Comment by soonahn** [comment]
@beghbali @cobbr2 @kenberland

File: lib/ruote/sequel/storage.rb [hunk 1]
```diff
@@ -454,9 +462,10 @@ def select_last_revs(docs)
     #
     # in order to cut down the number of selects, do one select with
     # all the information the worker needs for one step of work
-    #++
+    #
 
     CACHED_TYPES = %w[ msgs schedules configurations variables ]
+    NON_SCHEDULES = %w[ msgs configurations variables ]
```
Comment (007) [inline lib/ruote/sequel/storage.rb:468]: What's this for?

File: lib/ruote/sequel/storage.rb [hunk 2]
```diff
@@ -465,6 +474,7 @@ def prepare_cache
 
       CACHED_TYPES.each { |t| cache[t] = {} }
 
+
```
Comment (007) [inline lib/ruote/sequel/storage.rb:477]: Revert

**Comment by 007** [comment]
LGTM after `NON_SCHEDULES` removal

**Comment by kenberland** [comment]
Give me some context on what strategy you've settled upon.  These two PRs implement the whole thing?

**Comment by soonahn** [comment]
Using the `at` strategy with Mettraux blessing https://github.com/jmettraux/ruote-sequel/pull/13#issuecomment-173370851

This means extracting the `at` column, and using this newly indexed column to select the required `doc`'s from only the `schedules`.  I have some benchmarks and tests that simulate this if you'd like to see.

**Comment by soonahn** [comment]
This is only the first story in a two part epic.  I broke it up so at least I could have a checkpoint and some sort of mesaurable commit.

**Pull request closed without merging**